src/components/Contacts/ContactDetails/ContactDetailsTab/People/Items/PersonModal/PersonShowMore/PersonShowMore.tsx

```diff
@@ -58,11 +65,44 @@ export const PersonShowMore: React.FC<PersonShowMoreProps> = ({
     setFieldValue,
   } = formikProps;
 
+  useEffect(() => {
+    if (
+      typeof anniversaryMonth !== 'number' ||
+      typeof anniversaryDay !== 'number'
+    ) {
+      return;
+    }
+
+    const date = validateAndFormatInvalidDate(
+      anniversaryYear,
+      anniversaryMonth,
+      anniversaryDay,
+      locale,
+    );
+
+    setBackupAnniversaryDate(
+      date.formattedInvalidDate as unknown as DateTime<boolean>,
+    );
+    if (date.dateTime.invalidExplanation) {
+      setAnniversaryDateIsInvalid(true);
+    }
+  }, [anniversaryMonth, anniversaryDay, anniversaryYear]);
+
   const handleDateChange = (date: DateTime | null) => {
-    setFieldValue('anniversaryDay', date?.day || null);
-    setFieldValue('anniversaryMonth', date?.month || null);
-    setFieldValue('anniversaryYear', date?.year || null);
+    setFieldValue('anniversaryDay', date?.day ?? null);
+    setFieldValue('anniversaryMonth', date?.month ?? null);
+    setFieldValue('anniversaryYear', date?.year ?? null);
   };
+
+  const anniversaryDate =
+    anniversaryMonth && anniversaryDay
+      ? DateTime.local(
+          anniversaryYear ?? 1900,
+          anniversaryMonth,
+          anniversaryDay,
+        )
+      : null;
```

Comment:
This logic is still duplicated. Seems a simple function to extract and use:

```
const buildDate = (month, day, year) => {
  return month && day ? DateTime.local(year ?? 1900, month, day) : null;
};
```

Reply:
Ah, I worked on all the other duplicated parts but this one. I fixed it.